hbase_sql/src/main/java/com/splicemachine/ddl/ZooKeeperDDLWatchChecker.java

```diff
@@ -159,4 +189,30 @@ public void killDDLTransaction(String key){
        zkClient.deleteChangeNode(key);
     }
 
+
+    @Override
+    public void assignDDLDemarcationPoint(long txnId) throws Exception {
+        String rootPath = SIDriver.driver().getConfiguration().getSpliceRootPath();
+        String demarcationPath = rootPath+HConfiguration.DDL_PATH+"/"+id;
+        byte[] data = ZkUtils.getData(demarcationPath);
+        long demarcationPoint = data != null ? Bytes.toLong(data) : 0;
+        if (txnId > demarcationPoint) {
+            data = Bytes.toBytes(txnId);
+            ZkUtils.setData(demarcationPath, data, -1);
+            if (LOG.isDebugEnabled()) {
+                SpliceLogUtils.debug(LOG, "Set DDL demarcation point to %d", txnId);
+            }
+        }
+    }
+
+    @Override
+    public long getDemarcationPoint() throws IOException{
+        String rootPath = SIDriver.driver().getConfiguration().getSpliceRootPath();
+        String demarcationPointPath = rootPath+HConfiguration.DDL_PATH+"/"+id;
+        byte[] data = ZkUtils.getData(demarcationPointPath);
+        long demarcationPoint = data != null ? Bytes.toLong(data) : 0;
+        SpliceLogUtils.info(LOG, "Get DDL demarcation point %d from zookeeper", demarcationPoint);
+
+        return demarcationPoint;
+    }
```

Comment:
I'd just return the locally stored version from the init method

Reply:
This is invoked during initialization, when there is no local copy. I renamed it to initDemarcationPoint